lib/auth/db.go

```diff
@@ -43,14 +43,14 @@ func (s *Server) GenerateDatabaseCert(ctx context.Context, req *proto.DatabaseCe
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	hostCA, err := s.GetCertAuthority(types.CertAuthID{
-		Type:       types.HostCA,
+	databaseCA, err := s.GetCertAuthority(types.CertAuthID{
+		Type:       types.DatabaseCA,
 		DomainName: clusterName.GetClusterName(),
 	}, true)
```

Comment:
We need a fallback to HostCA here. In general, the proper upgrade procedure is to upgrade auth servers first but in case there's a newer database agent connects to an older auth server, this will fail. It seems easy enough to do a fallback here.

Reply:
Added